src/librbd/io/ObjectRequest.cc

```diff
@@ -223,7 +223,12 @@ void ObjectReadRequest<I>::read_object() {
   I *image_ctx = this->m_ictx;
 
   std::shared_lock image_locker{image_ctx->image_lock};
-  auto read_snap_id = this->m_io_context->read_snap().value_or(CEPH_NOSNAP);
+  librados::snap_t read_snap_id;
+  if ((m_read_flags & READ_FLAG_READ_OBJECT_HEAD_VERSION) != 0) {
```

Comment:
The only place where this new flag was used was on the write path, so how did the `io_context` have a value other than `CEPH_NOSNAP` already?

Reply:
hmmm actually this fix was just to be sure that we read the head version.
You know better than me if it's possible or not.
Should I remove it?

Reply:
It shouldn't be possible -- and since we also pass the `io_context` via the `ReadObjectRequest`, there should already be a way to set the `read_snap_id` by just passing your own `io_context` to the state machine.

Reply:
I removed this commit